code/modules/discord/discord_sql_functions.dm

```diff
@@ -0,0 +1,200 @@
+ *
+ * Arguments:
+ * * one_time_token the string of words representing the one time token
+ * * timebound A boolean flag, that specifies if it should only look for entries within the last 4 hours, false by default
+ *
+ * Returns a [/datum/discord_link_record]
+ */
+/client/proc/find_discord_link_by_token(one_time_token, timebound = FALSE)
+	var/timeboundsql = ""
+	if(timebound)
+		timeboundsql = "AND timestamp >= Now() - INTERVAL 4 HOUR"
+
+	var/query = {"
+		SELECT
+			ckey,
+			CAST(discord_id AS CHAR(25)),
+			timestamp,
+			one_time_token,
+			valid
+		FROM [format_table_name("discord_links")]
+		WHERE
+			one_time_token = :one_time_token
+			[timeboundsql]
+		ORDER BY timestamp DESC
+		LIMIT 1
+	"}
+
+	var/datum/db_query/query_get_discord_link_record = SSdbcore.NewQuery(
+		query,
+		list("one_time_token" = one_time_token)
+	)
+
+	if(!query_get_discord_link_record.Execute())
+		qdel(query_get_discord_link_record)
+		return
+
+	if(query_get_discord_link_record.NextRow())
+		var/result = query_get_discord_link_record.item
+		. = new /datum/discord_link_record(result[1], result[2], result[3], result[4], result[5])
+
+	//Make sure we clean up the query
+	qdel(query_get_discord_link_record)
+
+/**
+ * Find discord link entry by the passed in user ckey
+ *
+ * This will look into the discord link table and return the last (most recent) entry that matches the given ckey
+ *
+ * Arguments:
+ * * ckey the users ckey as a string
+ * * timebound A boolean flag, that specifies if it should only look for entries within the last 4 hours, false by default
+ *
+ * Returns a [/datum/discord_link_record]
+ */
+/client/proc/find_discord_link_by_ckey(ckey, timebound = FALSE)
+	var/timeboundsql = ""
+	if(timebound)
+		timeboundsql = "AND timestamp >= Now() - INTERVAL 4 HOUR"
+
+	var/query = {"
+		SELECT
+			ckey,
+			CAST(discord_id AS CHAR(25)),
+			timestamp,
+			one_time_token,
+			valid
+		FROM [format_table_name("discord_links")]
+		WHERE
+			ckey = :ckey
+			[timeboundsql]
+		ORDER BY timestamp DESC
+		LIMIT 1
+	"}
+
+	var/datum/db_query/query_get_discord_link_record = SSdbcore.NewQuery(
+		query,
+		list("ckey" = ckey)
+	)
+
+	if(!query_get_discord_link_record.Execute())
+		qdel(query_get_discord_link_record)
+		return
+
+	if(query_get_discord_link_record.NextRow())
+		var/result = query_get_discord_link_record.item
+		. = new /datum/discord_link_record(result[1], result[2], result[3], result[4], result[5])
+
+	//Make sure we clean up the query
+	qdel(query_get_discord_link_record)
+
+/**
+ * Find discord link entry by the passed in user ckey
+ *
+ * This will look into the discord link table and return the last (most recent) entry that matches the given ckey
+ *
+ * Arguments:
+ * * discord_id The users discord id (string)
+ * * timebound should we search only in the last 4 hours
+ *
+ * Returns a [/datum/discord_link_record]
+ */
+/client/proc/find_discord_link_by_discord_id(discord_id, timebound = FALSE)
```

Comment:
```suggestion
/**
 * Find discord link entry by the passed in user discord id
 *
 * This will look into the discord link table and return the last (most recent) entry that matches the given id number
 *
 * Arguments:
 * * discord_id The users discord id (string)
 * * timebound should we search only in the last 4 hours
 *
 * Returns a [/datum/discord_link_record]
 */
/client/proc/find_discord_link_by_discord_id(discord_id, timebound = FALSE)
```